internal/cmd/operator-sdk/pkgManToBundle/cmd.go

```diff
@@ -0,0 +1,294 @@
+    ├── 0.0.2
+    │   ├── etcdbackup.crd.yaml
+    │   ├── etcdcluster.crd.yaml
+    │   ├── etcdoperator.v0.0.2.clusterserviceversion.yaml
+    │   └── etcdrestore.crd.yaml
+    └── etcd.package.yaml
+
+# Run the following command to generate bundles in the default 'bundle/' directory with the base-container image name
+# to be 'quay.io/example/etcd'
+$ operator-sdk pkgman-to-bundle packagemanifests --base-image quay.io/example/etcd
+INFO[0000] packagemanifests will be migrated to bundles in bundle directory
+INFO[0000] Creating bundle/bundle-0.0.1/bundle.Dockerfile
+INFO[0000] Creating bundle/bundle-0.0.1/metadata/annotations.yaml
+...
+
+# After running the above command, the bundles will be generated in 'bundle/' directory.
+$ tree bundle/
+bundle/
+├── bundle-0.0.1
+│   ├── bundle.Dockerfile
+│   ├── manifests
+│   │   ├── etcdcluster.crd.yaml
+│   │   ├── etcdoperator.clusterserviceversion.yaml
+│   ├── metadata
+│   │   └── annotations.yaml
+│   └── tests
+│       └── scorecard
+│           └── config.yaml
+└── bundle-0.0.2
+    ├── bundle.Dockerfile
+    ├── manifests
+    │   ├── etcdbackup.crd.yaml
+    │   ├── etcdcluster.crd.yaml
+    │   ├── etcdoperator.v0.0.2.clusterserviceversion.yaml
+    │   ├── etcdrestore.crd.yaml
+    ├── metadata
+    │   └── annotations.yaml
+    └── tests
+        └── scorecard
+			└── config.yaml
+
+Also, images for the both the bundles will be built with the following names: quay.io/example/etcd:0.0.1 and quay.io/example/etcd:0.0.2.
+`
+)
+
+type pkgManToBundleCmd struct {
+	// Input packagemanifest directory.
+	pkgmanifestDir string
+
+	// Optional flags for generating and building bundles.
+	outputDir string
+	baseImg   string
+	buildCmd  string
+}
+
+// NewCmd returns the pkgManToBundleCmd configured with the provided input options.
+func NewCmd() *cobra.Command {
+	p := pkgManToBundleCmd{}
+
+	pkgManToBundleCmd := &cobra.Command{
+		Use:     "pkgman-to-bundle",
+		Short:   "Migrates packagemanifests to bundles",
+		Long:    longHelp,
+		Example: examples,
+		PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			return p.validate(args)
+		},
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			p.pkgmanifestDir = args[0]
+			return p.run()
+		},
+	}
+
+	pkgManToBundleCmd.Flags().StringVar(&p.outputDir, "output-dir", "", "Directory to write bundle to.")
+	pkgManToBundleCmd.Flags().StringVar(&p.baseImg, "base-image", "", "Base container image name for bundles.")
+	pkgManToBundleCmd.Flags().StringVar(&p.buildCmd, "build-cmd", "", "Build command to be run for building images. By default 'docker build' is run.")
+
+	return pkgManToBundleCmd
+}
+
+// Generate the bundles from the provided packagemanifest directory.
+func (p *pkgManToBundleCmd) run() (err error) {
+	p.setDefaults()
+
+	// remove directories if they exist previously.
+	if _, err := os.Stat(p.outputDir); !os.IsNotExist(err) {
+		// TODO(Verify): Do we need an overwrite flag here ?
```

Comment:
Instead of deleting this directory, the command should error out and let the user handle cleanup themselves. The less hidden side effects, the better.